### PR TITLE
Add accelerated networking flag to NIC update/create

### LINKED ISSUE #3749: [Network] Add support for accelerated networking to NIC create/update
Swagger Spec: https://github.com/Azure/azure-rest-api-specs/blob/master/arm-network/2016-09-01/swagger/networkInterface.json
Flag: enableAcceleratedNetworking

### PR DESCRIPTION
**Pull request by tjprescott**
Closes #3749.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] The PR has modified HISTORY.rst with an appropriate description of the change (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

### Command Guidelines

- [x] Each command and parameter has a meaningful description.
- [N/A] Each new command has a test.

(see [Authoring Command Modules](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules))

### CODE REVIEW

File: src/command_modules/azure-cli-network/HISTORY.rst [hunk 1]
```diff
@@ -2,8 +2,14 @@
 
 Release History
 ===============
+
+unreleased
+++++++++++++++++++
+* `nic create/update`: Add support for `--accelerated-networking`.
+* `nic create`: Remove non-functional `--internal-dns-name-suffix` argument.
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-network/HISTORY.rst]: This should be marked as `BC`, though i am fine we can release w/o deprecating process

File: src/command_modules/azure-cli-network/azure/cli/command_modules/network/custom.py [hunk 2]
```diff
@@ -1002,13 +1002,12 @@ def set_lb_rule(
...
 def create_nic(resource_group_name, network_interface_name, subnet, location=None, tags=None,
-               internal_dns_name_label=None, dns_servers=None,
-               internal_domain_name_suffix=None, enable_ip_forwarding=False,
+               internal_dns_name_label=None, dns_servers=None, enable_ip_forwarding=False,
                load_balancer_backend_address_pool_ids=None,
                load_balancer_inbound_nat_rule_ids=None,
                load_balancer_name=None, network_security_group=None,
                private_ip_address=None, private_ip_address_version=None,
-               public_ip_address=None, virtual_network_name=None):
+               public_ip_address=None, virtual_network_name=None, enable_accelerated_networking=None):
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-network/azure/cli/command_modules/network/custom.py]: Can we update existing tests to cover this new flag? Or it is hard?

Comment (tjprescott) [inline src/command_modules/azure-cli-network/azure/cli/command_modules/network/custom.py]: Nope, it's simple. I'll add it.

**Comment by tjprescott** [comment]
@yugangw-msft comments addressed.

**Pull request merged**